providers/ms365/resources/applications.go

```diff
@@ -176,18 +176,50 @@ func newMqlMicrosoftApplication(runtime *plugin.Runtime, app models.Applicationa
 }
 
 func initMicrosoftApplication(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
-	// we only look up the application if we have been supplied by its name and nothing else
+	var appId string
+	// check if we want to lookup by name or id
 	raw, ok := args["name"]
-	if !ok || len(args) != 1 {
-		return args, nil, nil
+	if ok && len(args) == 1 {
+		name := raw.Value.(string)
+		id, err := fetchApplicationIdByName(runtime, name)
+		if err != nil {
+			return nil, nil, err
+		}
+		appId = *id
 	}
-	name := raw.Value.(string)
+	if appId == "" {
+		raw, ok := args["id"]
```

Comment:
should id take precedence if both are specified

Reply:
right now we're assuming that only one will be provided/are rejecting if more than one provided. but ya, if we got both it would be better it would be better to use id